pycqed/utilities/general.py

```diff
@@ -419,3 +421,22 @@ def setInDict(dataDict: dict, mapList: list, value):
                         'b': 4}
     """
     getFromDict(dataDict, mapList[:-1])[mapList[-1]] = value
+
+
+class NumpyJsonEncoder(json.JSONEncoder):
+    '''
+    JSON encoder subclass that converts Numpy types to native python types
+    for saving in JSON files.
+    Also converts datetime objects to strings.
+    '''
+    def default(self, o):
+        if isinstance(o, np.integer):
+            return int(o)
+        elif isinstance(o, np.floating):
+            return float(o)
+        elif isinstance(o, np.ndarray):
+            return o.tolist()
+        elif isinstance(o, datetime.datetime):
```

Comment:
Should we add a JSON decoder to be able to extract this or should we just accept that as one of the limitations of JSON?